hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -147,23 +184,21 @@ private void initIfNeeded() {
         }
       }
       timings.add(timer.endTimer());
-      LOG.info(String.format("Metadata read for key %s took [open, baseFileRead, logMerge] %s ms", key, timings));
+      LOG.info(String.format("Metadata read for key %s took [baseFileRead, logMerge] %s ms", key, timings));
       return Option.ofNullable(hoodieRecord);
     } catch (IOException ioe) {
       throw new HoodieIOException("Error merging records from metadata table for key :" + key, ioe);
-    } finally {
-      closeIfNeeded();
     }
   }
 
   /**
-   * Open readers to the base and log files.
+   * Returns the readers to the base and log files.
+   *
+   * If reuse is allowed then cached readers are returned. Otherwise new readers are opened.
```

Comment:
please correct me if I am wrong. within this method, we don't care about reuse enabled or not. So, wondering if we need to remove this comment?

Reply:
Are these addressed? can you please respond.

Reply:
Fixed the comment.